flink-ai-flow/ai_flow/java/client/src/test/java/com/aiflow/client/AIFlowClientTest.java

```diff
@@ -0,0 +1,779 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.aiflow.client;
+
+import com.aiflow.common.DataType;
+import com.aiflow.common.ModelStage;
+import com.aiflow.common.Status;
+import com.aiflow.entity.ArtifactMeta;
+import com.aiflow.entity.DatasetMeta;
+import com.aiflow.entity.ModelMeta;
+import com.aiflow.entity.ModelRelationMeta;
+import com.aiflow.entity.ModelVersion;
+import com.aiflow.entity.ModelVersionMeta;
+import com.aiflow.entity.ModelVersionRelationMeta;
+import com.aiflow.entity.ProjectMeta;
+import com.aiflow.entity.RegisteredModel;
+import com.aiflow.entity.WorkflowMeta;
+import com.aiflow.exception.AIFlowException;
+import com.aiflow.notification.client.EventWatcher;
+import com.aiflow.notification.entity.EventMeta;
+import com.aiflow.proto.Message;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class AIFlowClientTest {
+
+    private static final String LOCALHOST = "localhost";
+    private static final int SEC = 1000;
+
+    private static AIFlowClient client;
+    private static final Map EMPTY_MAP = Collections.emptyMap();
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        int port = Integer.valueOf(getProperties("port"));
+        client = new AIFlowClient(LOCALHOST + ":" + port, "default",
+                false, 1 * SEC, 1 * SEC, 1 * SEC);
+    }
+
+    @AfterAll
+    public static void afterClass() throws Exception {
+        client.stopListenEvent("default", "");
+    }
+
+
+
+    private static String getProperties(String key) throws IOException {
+        InputStream in = new Object() {
+            public InputStream getInputStream() {
+                return this.getClass().getClassLoader().getResourceAsStream("test.properties");
+            }
+        }.getInputStream();
+        Properties properties = new Properties();
+        properties.load(in);
+        return properties.getProperty(key);
+    }
+
+    public static void main(String[] args) throws IOException {
+        System.out.println(getProperties("port"));
+    }
+
+    // test dataset
+
+    @Test
+    public void testRegisterDataset() throws Exception {
```

Comment:
It's recommended to combine the tests for the operations of dataset like `testModelOperations` etc.